src/theory/arith/arith_rewriter.cpp

```diff
@@ -749,8 +749,10 @@ RewriteResponse ArithRewriter::postRewriteIAnd(TNode t)
     }
     if (t[i].getConst<Rational>().getNumerator() == Integer(2).pow(bsize) - 1)
     {
-      // ((_ iand k) 111...1 y) ---> y
-      return RewriteResponse(REWRITE_DONE, t[i == 0 ? 1 : 0]);
+      // ((_ iand k) 111...1 y) ---> (mod y 2^k)
+      Node twok = nm->mkConstInt(Rational(Integer(2).pow(bsize)));
+      Node ret = nm->mkNode(kind::INTS_MODULUS, t[1-i],  twok);
```

Comment:
```suggestion
      Assert(i==0 || i==1);
      Node ret = nm->mkNode(kind::INTS_MODULUS, t[1-i],  twok);
```

Reply:
This is a pretty obvious assertion, i is being looped 0, <2 in the for-loop above :)